src/index.ts

```diff
@@ -0,0 +1,168 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import InstanceSkel from '../../../instance_skel';
+import { YoutubeConfig, listConfigFields, loadMaxBroadcastCount, loadRefreshInterval } from './config';
+import {
+	CompanionFeedbackEvent,
+	CompanionFeedbackResult,
+	CompanionSystem,
+	CompanionInputField,
+	CompanionActionEvent,
+} from '../../../instance_skel_types';
+import { YoutubeAuthorization } from './authFlow';
+import { Core, ModuleBase } from './core';
+import { handleFeedback, listFeedbacks } from './feedbacks';
+import { StateMemory, Broadcast } from './cache';
+import { getBroadcastVars, exportVars, declareVars } from './vars';
+import { listPresets } from './presets';
+import { listActions, handleAction } from './actions';
+
+/**
+ * Main Companion integration class of this module
+ */
+class YoutubeInstance extends InstanceSkel<YoutubeConfig> implements ModuleBase {
+	/** Executive core of the module */
+	private Core?: Core;
+	/** YouTube authorization flow */
+	private Auth: YoutubeAuthorization;
+
+	/**
+	 * Create a new instance of this module
+	 * @param system Companion internals
+	 * @param id Module ID
+	 * @param config Module configuration
+	 */
+	constructor(system: CompanionSystem, id: string, config: YoutubeConfig) {
+		super(system, id, config);
+		this.Auth = new YoutubeAuthorization(config, this);
+	}
+
+	/**
+	 * Initialize this module (i.e. authorize to YT, fetch data from it and initialize actions, feedbacks, etc.)
+	 * @param isReconfig Whether the initialization is done as part of module reconfiguration
+	 */
+	init(isReconfig = false): void {
+		this.log('debug', 'Initializing YT module');
+		this.status(this.STATUS_WARNING, 'Initializing');
+
+		this.Auth.authorize(isReconfig)
+			.then((googleAuth) => {
+				this.saveToken(JSON.stringify(googleAuth.credentials));
+
+				this.Core = new Core(this, googleAuth, loadMaxBroadcastCount(this.config), loadRefreshInterval(this.config));
+
+				this.Core.init()
+					.then(() => {
+						this.log('info', 'YT Module initialized successfully');
+						this.status(this.STATUS_OK);
+					})
+					.catch((err) => {
+						this.log('warn', `YT Broadcast query failed: ${err}`);
+						this.status(this.STATUS_ERROR, `YT Broadcast query failed: ${err}`);
+						this.Core?.destroy();
+						this.Core = undefined;
+					});
+			})
+			.catch((reason) => {
+				this.saveToken('');
+				this.log('warn', `Authorization failed: ${reason}`);
+				this.status(this.STATUS_ERROR, `Authorization failed: ${reason}`);
+			});
+	}
+
+	/**
+	 * Save an OAuth2 authorization token to the persistent settings store.
+	 * @param raw Stringified token or empty value
+	 */
+	saveToken(raw: string): void {
+		this.config.auth_token = raw;
+		this.system.emit('instance_config_put', this.id, this.config, true);
```

Comment:
saveConfig is now in the typescript interface, so you should switch back to using that.

Reply:
Thank you for adding it to Companion, I have changed the module to do `saveConfig()` now.